### PR TITLE
Enable encrypted connections with TLS

### PR DESCRIPTION
**Pull request by lorextera**
Added support to encrypted connections
`$redis = new Credis_Client('tls://hostname:port')`

### CODE REVIEW

File: Client.php [hunk 1]
```diff
@@ -421,6 +433,9 @@ protected function convertHost()
         if ($this->port !== NULL && substr($this->host,0,1) == '/') {
             $this->port = NULL;
```
Comment (Xon) [inline Client.php]: Need a `$this->scheme = 'unix';` here.

File: Client.php [hunk 2]
```diff
@@ -402,8 +412,9 @@ public function setCloseOnDestruct($flag)
     }
     protected function convertHost()
     {
-        if (preg_match('#^(tcp|unix)://(.*)$#', $this->host, $matches)) {
+        if (preg_match('#^(tcp|tls|unix)://(.*)$#', $this->host, $matches)) {
             if($matches[1] == 'tcp') {
```
Comment (Xon) [inline Client.php]: Missing a check against `$matches[1] == 'tls'`, so TLS is hitting the unix socket code path.

**Comment by Xon** [comment]
This really need some TLS tests as well, as `convertHost` is non-functional for a bunch of inputs

File: Client.php (hunk 2 above)
Comment (lorextera) [reply Client.php]: Yes, I forget if ($matches[1] == 'tcp' || $matches[1] == 'tls') {

**Pull request closed without merging**